python.md

```diff
@@ -0,0 +1,216 @@
+- Add a lab-specific `Side Effects` section, for any side
+  effects of your function. If you have doubts about whether your function has
+  side effects, consult with the reviewer.
+- Add other sections supported by the [Napoleon Sphinx
+  extension](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/#docstring-sections)
+  as you see fit.
+- `Returns` should be the last section.
+- Don't add empty sections.
+- Generally try to minimize vertical space by avoiding redundancy and
+  aiming for conciseness. It is desirable to fit docstring and function
+  body on a reasonably sized screen.
+- Use vanilla
+  [reStructuredText/Sphinx](https://www.sphinx-doc.org/en/master/usage/restructuredtext/index.html)
+  markup sparingly in order to enhance visual appeal on [*Read the
+  Docs*](https://docs.readthedocs.io/en/stable/), but without impeding
+  readability of raw docstrings. For instance, it is a good idea to mark up a
+  block of code using the rather subtle [double colon, blank line and
+  indentation](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks),
+  but not to litter a docstring with [Sphinx
+  directives](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html).
+
+
+#### TODO-comments
+*[cf. Google Python Style Guide [TODO Comments](
+https://google.github.io/styleguide/pyguide.html#312-todo-comments)]*
+
+Adding a parenthesized identifier in a `# TODO:`-comment is not necessary. The
+version control system can provide that information just as well.
+
+### Strings
+*[cf. Google Python Style Guide
+[Strings](https://google.github.io/styleguide/pyguide.html#310-strings)]*
+
+Use `f"...{var}"` when writing Python 3.6+ code, or `"...{0}".format(var)`
+anywhere. `"..." + var` may be used for simple cases, but beware of pitfalls
+such as easily missed whitespace, or `var` not being a string. Don't use the
+old `"...%s" % var`-notation.
+
+
+## Programming Recommendations
+
+### Miscellaneous
+
+<!-- TODO: This recommendation is an updated remainder of the old style guide.
+It seems a bit lost here. -->
+
+- Avoid `if` and `else` keywords be separated by a lot of code.
+  ```python
+  # NO:
+  if bizbaz:
+    frobnicate(flimflam)
+    ...
+    ...
+    ...
+
+  else:
+    raise YamException("No bizbaz, no frobnicate!")
+  ```
+  Use the *"return early"* pattern, if the `else` clause would raise an
+  exception or returns:
+
+  ```python
+  # YES:
+  if not bizbaz:
+    raise YamException("No bizbaz, no frobnicate!")
+
+  frobnicate(flimflam)
+  ...
+  ...
+  ...
+
+  ```
+  Returning early is generally useful to avoid excessive levels of nesting
+  (aka. *"arrowhead anti-pattern"*).
+
+  If returning early is not an option, and  `if` and `else` are thus both
+  needed, it is slightly preferable to keep the shorter clause first, so that
+  the `else` isn't too far from the `if`.
+
+
+
+### Testing
+
+<!-- TODO: Should we add more tangible instructions? -->
```

Comment:
This might be self-explanatory, but we could give advice about testing at least all paths through the function, etc

Reply:
Also, I will add to make sure that the test execution order doesn't matter.
Also, because most of the tests are unit tests, maybe it's worth mentioning that unit tests shouldn't be dependent on external service and if they do, then mocking could be a solution.

Reply:
Thanks. I decided to just remove this section to keep this document minimal. Let's think of more actionable recommendations in https://github.com/theupdateframework/tuf/pull/1129 (where I also referenced your suggestions here).